nilearn/glm/first_level/first_level.py

```diff
@@ -1903,18 +1911,18 @@ def _check_kwargs_load_confounds(**kwargs):
     }
 
     if kwargs.get("confounds_strategy") is None:
-        return None
-
-    kwargs_load_confounds = {
-        key: (
-            defaults[key]
-            if f"confounds_{key}" not in kwargs
-            else kwargs[f"confounds_{key}"]
-        )
-        for key in defaults
-    }
+        return None, kwargs
+
+    remaining_kwargs = kwargs.copy()
```

Comment:
'remaining_kwargs' should include only kwargs without the 'confounds' string, right ?

Reply:
Yes:
- If the kwarg is `confounds_`-something, then we effectively strip the `confounds_` prefix and replace the previously defined default value. This gets added to `kwargs_load_confounds`, to be passed to another function
-  Otherwise, the kwarg isn't removed and stays in `remaining_kwargs`. The reason why I am returning `remaining_kwargs` instead of directly raising an error is just in case we want to support other kwargs in the future